homeassistant/components/unifiprotect/select.py

```diff
@@ -349,9 +349,9 @@ def __init__(
         description: ProtectSelectEntityDescription,
     ) -> None:
         """Initialize the unifi protect select entity."""
+        self._async_set_options(data, description)
```

Comment:
Is there a reason to have this before super() ? Its at least confusing, does it manipulate data also manipulated in super().

Reply:
We need to set the following before calling the super init
```python
        self._attr_options = [item["name"] for item in options]
        self._hass_to_unifi_options = {item["name"]: item["id"] for item in options}
        self._unifi_to_hass_options = {item["id"]: item["name"] for item in options}
```

This is because the super init is going to do the first update and call `_async_update_device_from_protect` which will needs these values set already

```python
        self._attr_current_option = self._unifi_to_hass_options.get(
            unifi_value, unifi_value
        )
```
https://github.com/home-assistant/core/pull/100439/files#diff-5434e60375b221188d4bf9c3ab2959506d20f9a742a9870cc505864316ba9b3dR372